model/src/rollover.rs

```diff
@@ -84,3 +99,167 @@ impl RolloverParams {
+    pub settlement_event_id: BitMexPriceEventId,
+    pub revocation_sk_ours: SecretKey,
+    pub complete_fee: CompleteFee,
+}
+
+impl Dlc {
+    pub fn base_dlc_params(
+        &self,
+        base_commit_txid: Txid,
+        complete_fee: CompleteFee,
+    ) -> Result<BaseDlcParams> {
+        Ok(if self.commit.0.txid() == base_commit_txid {
+            self.base_dlc_params_from_latest(complete_fee)
+        } else {
+            let params = self.base_dlc_params_from_revoked_commit(base_commit_txid)?;
+            tracing::info!(commit_txid=%base_commit_txid, "Building base DLC params for rollover from previous commit TX");
+
+            params
+        })
+    }
+
+    /// Construct the `BaseDlcParams` based on the latest DLC.
+    ///
+    /// This is called if neither party has fallen behind in terms of
+    /// rollovers.
+    pub fn base_dlc_params_from_latest(&self, complete_fee: CompleteFee) -> BaseDlcParams {
+        BaseDlcParams {
+            base_commit_params: BaseCommitParams {
+                commit_encsig_ours: self.commit.1,
+                revocation_pk_theirs: self.revocation_pk_counterparty,
+                publish_pk_theirs: self.publish_pk_counterparty,
+                commit_txid: self.commit.0.txid(),
+                commit_script_pubkey: self.commit.2.script_pubkey(),
+                settlement_event_id: self.settlement_event_id,
+                revocation_sk_ours: self.revocation,
+                complete_fee,
+            },
+            revoked_commits: self.revoked_commit.clone(),
+        }
+    }
+
+    /// Construct the `BaseDlcParams` based on a revoked DLC. The
+    /// commit TXID is used to identify the rollover from which both
+    /// parties intend to start the new rollover.
+    ///
+    /// This is called if a party is behind in terms of rollovers.
+    fn base_dlc_params_from_revoked_commit(
+        &self,
+        revoked_commit_txid: Txid,
+    ) -> Result<BaseDlcParams> {
+        let RevokedCommit {
+            encsig_ours: commit_encsig_ours,
+            revocation_sk_ours,
+            revocation_sk_theirs,
+            publish_pk_theirs,
+            txid: commit_txid,
+            script_pubkey: commit_script_pubkey,
+            settlement_event_id,
+            complete_fee,
+        } = self
+            .revoked_commit
+            .iter()
+            .find(|revoked_commit| revoked_commit.txid == revoked_commit_txid)
+            .with_context(|| format!("Unknown commit TXID {revoked_commit_txid}"))?
+            .clone();
+
+        let settlement_event_id = settlement_event_id.context("Missing settlement event id")?;
+        let complete_fee = complete_fee.context("Missing complete fee")?;
+        let revocation_sk_ours = revocation_sk_ours.context("Missing own revocation sk")?;
+
+        let revoked_commits = self
+            .revoked_commit
+            .iter()
+            .take_while(|commit| commit.txid != revoked_commit_txid)
```

Comment:
indeed, perhaps a simple `filter` might be better here as it doesn't need such assumption.

Reply:
Once https://github.com/itchysats/itchysats/pull/2478 merges it should be guaranteed. I will see to fix and merge that one.

Reply:
It's a good point, but I don't see any reason why it shouldn't be sorted, we just append to it during rollover.

It would be nice if could just call `sort_by` on the iterator, but we need the `settlement_id`, which is optional for backwards compatibility.